meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,11 +41,98 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+
+		importVariants(event: React.ChangeEvent<HTMLInputElement>) {
+			const { t } = this.props
+
+			const file = event.target.files ? event.target.files[0] : null
+			if (!file) {
+				return
 			}
+
+			const reader = new FileReader()
+			reader.onload = (progressEvent: ProgressEvent<FileReader>) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (progressEvent.target as any).result
```

Comment:
`as any` is discouraged. Instead of using the progressEvent, you can use `reader.result`. Since the type is `string | ArrayBuffer` and we only use `reader.readAsText` we can safely use the type assertion `reader.result as string`.

And as @LindvedKrvang already mentioned, `uploadFileContents` semantically indicates that it contains a function.

```suggestion
			reader.onload = () => {
				this.setState({
					uploadFileKey: Date.now(),
				})

				const fileContents = reader.result as string
```

One could consider making another listener for the reader in case it fails reading the file and notifying the user.

Reply:
Added it to [f74f966](https://github.com/tv2/tv-automation-server-core/pull/188/commits/f74f9665479993af5c8647b78f40a11fe9370e0f).